### PR TITLE
Feature/kyv 467 subscription payment failed email

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: packages/order-experience-api/src/api/sendSubscriptionPaymentFailedEmail.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    await validateApiKey({ apiKey, namespace: 'admin' })
+    // Assumption: subscription is currently on its first order
+    const subscription = await getSubscriptionAdmin({ id })
+    const order = await getOrderAdmin({ orderId: subscription.orderId })
+
+    await sendSubscriptionPaymentFailedEmailToCustomer({
+      order: order,
+      subscription: subscription,
+      sendTo: subscription.customerEmail,
+      emailHeader: 'Maksukortin veloitus epäonnistui',
```
Comment (severikupari1) [inline packages/order-experience-api/src/api/sendSubscriptionPaymentFailedEmail.ts:42]: I will add other translations to this

File: packages/order-experience-api/tsconfig.json [hunk 2]
```diff
@@ -29,9 +29,6 @@
             ],
             "@verkkokauppa/auth-helsinki-profile": [
                 "../auth-helsinki-profile"
-            ],
-            "@verkkokauppa/message-backend": [
-                "../message-backend"
```
Comment (severikupari1) [inline packages/order-experience-api/tsconfig.json:34]: I will check why these are deleted.

File: packages/message-backend/src/email/send/service.ts [hunk 3]
```diff
@@ -74,21 +89,30 @@ export const sendEmail = async (p: {
...
     return res.data
   } catch (e) {
-    throw new SendOrderConfirmationEmailFailure(e)
+    switch (p.emailType) {
+      case 'orderConfirmation':
+        throw new SendOrderConfirmationEmailFailure(e)
+      case 'subscriptionPaymentFailed':
+        throw new SendSubscriptionPaymentFailedEmailToCustomer(e)
+      case 'subscriptionContract':
+        throw new SendSubscriptionPaymentFailedEmailToCustomer(e)
```
Comment (severikupari1) [inline packages/message-backend/src/email/send/service.ts:98]: I will change this to correct error

File: packages/order-backend/src/service.spec.ts [hunk 4]
```diff
@@ -125,6 +128,7 @@ describe('Test Create Order', () => {
       priceVat: '24',
       priceTotal: '124',
       items: mockData.items,
+      updateCardUrl: `${process.env.CHECKOUT_BASE_URL}${mockData.order.orderId}/update-card?user=${mockData.order.user}`,
       checkoutUrl: `https://checkout.dev.hel/${mockData.order.orderId}`,
```
Comment (eerohakio) [inline packages/order-backend/src/service.spec.ts:132]: These could be used with CHECKOUT_BASE_URL as well

File: tsconfig.json [hunk 5]
```diff
@@ -38,9 +38,6 @@
             ],
             "@verkkokauppa/message-experience-api/*": [
                 "message-experience-api/src/*"
-            ],
-            "@verkkokauppa/auth-helsinki-profile/*": [
-                "auth-helsinki-profile/src/*"
```
Comment (eerohakio) [inline tsconfig.json:43]: Why is this removed? Not required anymore?

Comment (severikupari1) [reply tsconfig.json:43]: Good catch, I had some problems with these tsconfig files because windows changes the paths and I need to reset these files manually.

**Review by eerohakio — APPROVED** [review]
